tests/e2e/default/enketo/training-cards.wdio-spec.js

```diff
@@ -45,10 +43,56 @@ describe('Training Cards', () => {
     await trainingCardsPage.waitForTrainingCards();
 
     const confirmMessage = await trainingCardsPage.quitTraining();
-    expect(confirmMessage.header).to.equal('Leave training?');
+    expect(confirmMessage.header).to.equal(expectedConfirmModalHeader);
+    expect(confirmMessage.body).to.contain(expectedConfirmMessage);
+    await trainingCardsPage.confirmQuitTraining();
+    await trainingCardsPage.checkTrainingCardIsNotDisplayed();
+
+    await commonPage.goToReports();
+    await commonElements.waitForPageLoaded();
+    expect(await reportsPage.leftPanelSelectors.allReports()).to.be.empty;
+  });
+
+  it('should display confirm message before navigating to different page', async () => {
+    await browser.refresh();
+    await trainingCardsPage.waitForTrainingCards();
+
+    await commonPage.goToPeople();
+
+    const confirmMessage = await modalPage.getModalDetails();
+    expect(confirmMessage.header).to.contain(expectedConfirmModalHeader);
     expect(confirmMessage.body).to.contain(expectedConfirmMessage);
+    expect(await browser.getUrl()).to.contain('/reports');
```

Comment:
By doing this validation, you are trusting that the previous `it` passed fine and that the test is in the reports section. But if for some reason the previous test failed and it was not able to access the reports section, this test will also fail.

We try to make every test completely independent, especially because we do retries when a test fails. Please add at the beginning of the test to go to Reports, in that case, you would not depend on the previous test. Or maybe add a `beforeEach` hook to go to Reports. Whatever works best 😊